pkg/noun/jets/b/mate.c

```diff
@@ -0,0 +1,30 @@
+/// @file
+
+#include "jets/q.h"
+#include "jets/w.h"
+
+#include "noun.h"
+
+
+  u3_noun
+  u3qb_mate(u3_noun a,
+            u3_noun b)
+  {
+    if ( u3_nul == b ) {
+      return u3k(a);
+    } else if ( u3_nul == a ) {
+      return u3k(b);
+    } else if ( c3y == u3r_sing(a, b) ) {
+      return u3k(a);
+    } else {
+      return u3m_bail(c3__fail);
```

Comment:
You can match the trace exactly here with `u3m_error("mate");`

Reply:
I now see something like this:

```
> (need ~)
dojo: hoon expression failed
> (mate `4 `5)
mate
dojo: hoon expression failed
```

Is there any way to omit the extra message? I also tried `u3m_error("")` and `u3m_error(u3_nul)`, but to no avail.

Reply:
Sorry i missed this. The extra message is consistent with the hoon/nock for +mate. Arguably that's too verbose, but I don't think anyone has complained yet.

To produce a deterministic error without including "mate" in the stack trace, you'd just call `u3m_bail(c3__exit);`. Notably, the hoon for `+need` includes an error message, but the jet does not.

I try to keep jets consistent wrt to error messages, but system doesn't enforce that to the degree it should, and it's much less of a problem than any other kind of mismatch.